src/adapters/supabase/helpers/wallet.ts

```diff
@@ -15,8 +15,11 @@ export class Wallet extends Super {
       throw error;
     }
 
-    console.info("Successfully fetched wallet", { userId, address: data.wallets?.address });
-    return data.wallets?.address;
+    // Check if wallets is an array, if so, return the first element's address
+    const address = Array.isArray(data.wallets) ? data.wallets[0]?.address : data.wallets?.address;
```

Comment:
There should only be a single wallet associated per user. Why did you check for arrays

Reply:
@whilefoo @rndquu can you concur this is correct?

Reply:
Can't you use `single()` to retrieve only one? The type should be properly deduced.

Reply:
yes it's normal for that to be an array because it's querying an user with wallets as a relation. if you wanted to get just one record you would need to query the wallet table directly

Reply:
@whilefoo The code is already in the base branch but isn't working without the changes in this PR. Should I move the hotfix to a separate PR?